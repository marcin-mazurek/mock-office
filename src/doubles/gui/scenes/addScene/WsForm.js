import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { init as initAddScene } from './actions';
import { getSelectedServerDetails } from '../../entities/servers/selectors';

export class WsForm extends React.Component {
  constructor(props) {
    super(props);

    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleIntervalChange = this.handleIntervalChange.bind(this);
    this.handleRequirementsChange = this.handleRequirementsChange.bind(this);
    this.handleDelayChange = this.handleDelayChange.bind(this);
    this.state = {};
  }

  handleIntervalChange(event) {
    this.setState({
      interval: event.target.value
    });
  }

  handleRequirementsChange(event) {
    this.setState({
      requirements: event.target.value
    });
  }

  handleDelayChange(event) {
    this.setState({
      delay: event.target.value
    });
  }

  handleSubmit(event) {
    event.preventDefault();
    const delay = parseInt(this.state.dely, 10);
    const interval = parseInt(this.state.interval, 10);

    let requirements;

    try {
      if (this.state.requirements) {
        requirements = JSON.parse(this.state.requirements);
      }
    } catch (error) {
      // eslint-disable-next-line no-console
      console.error(error.message);
    }

    this.props.initAddScene(this.props.queueId, {
      requirements,
      delay,
      interval
    });
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <div>
          <label htmlFor="interval">Interval:</label>
          <input type="number" name="interval" onChange={this.handleIntervalChange} />
        </div>
        <div>
          <label htmlFor="delay">Delay:</label>
          <input type="number" name="delay" onChange={this.handleDelayChange} />
        </div>
        <div>
          <label htmlFor="requirements">Requirements:</label>
          <div>
            <textarea
              name="requirements"
              cols="30"
              rows="10"
              onChange={this.handleRequirementsChange}
            />
          </div>
        </div>
        <div>
          <label htmlFor="payload">Payload:</label>
          <div>
            <textarea name="payload" cols="30" rows="10" onChange={this.handlePayloadChange} />
          </div>
        </div>
        <div>
          <button type="submit">
            Submit
          </button>
        </div>
      </form>
    );
  }
}

WsForm.propTypes = {
  initAddScene: PropTypes.func.isRequired,
  queueId: PropTypes.string.isRequired
};

const mapStateToProps = state => ({
  queueId: getSelectedServerDetails(state).queue
});

const mapDispatchToProps = {
  initAddScene
};

export default connect(mapStateToProps, mapDispatchToProps)(WsForm);
