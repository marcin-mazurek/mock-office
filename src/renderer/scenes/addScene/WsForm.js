import React from 'react';
import { connect } from 'react-redux';
import { init as initAddScene } from './actions';
import { getSelectedServerDetails } from '../../servers/selectors';

export class WsForm extends React.Component {
  constructor(props) {
    super(props);

    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleIntervalChange = this.handleIntervalChange.bind(this);
    this.handleBlockingChange = this.handleBlockingChange.bind(this);
    this.handleRequirementsChange = this.handleRequirementsChange.bind(this);
    this.handlePayloadChange = this.handlePayloadChange.bind(this);
    this.handleDelayChange = this.handleDelayChange.bind(this);
    this.state = {};
  }

  handleIntervalChange(event) {
    this.setState({
      interval: event.target.value
    });
  }

  handleBlockingChange(event) {
    this.setState({
      blocking: event.target.checked
    });
  }

  handleRequirementsChange(event) {
    this.setState({
      requirements: event.target.value
    });
  }

  handlePayloadChange(event) {
    this.setState({
      scenePayload: event.target.value
    });
  }

  handleDelayChange(event) {
    this.setState({
      delay: event.target.value
    });
  }

  handleSubmit(event) {
    event.preventDefault();
    const delay = parseInt(this.state.dely, 10);
    const interval = parseInt(this.state.interval, 10);

    let requirements;

    try {
      if (this.state.requirements) {
        requirements = JSON.parse(this.state.requirements);
      }
    } catch (error) {
      // eslint-disable-next-line no-console
      console.error(error.message);
    }

    let scenePayload;

    try {
      scenePayload = JSON.parse(this.state.scenePayload);
    } catch (error) {
      // eslint-disable-next-line no-console
      console.error(error.message);
    }

    this.props.initAddScene(this.props.queueId, {
      requirements,
      delay,
      scenePayload,
      blocking: this.state.blocking,
      interval
    });
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <div>
          <label htmlFor="interval">Interval:</label>
          <input type="number" name="interval" onChange={this.handleIntervalChange} />
        </div>
        <div>
          <label htmlFor="blocking">Blocking:</label>
          <input
            type="checkbox"
            name="blocking"
            onChange={this.handleBlockingChange}
            value={this.state.blocking}
          />
        </div>
        <div>
          <label htmlFor="delay">Delay:</label>
          <input type="number" name="delay" onChange={this.handleDelayChange} />
        </div>
        <div>
          <label htmlFor="requirements">Requirements:</label>
          <div>
            <textarea
              name="requirements"
              cols="30"
              rows="10"
              onChange={this.handleRequirementsChange}
            />
          </div>
        </div>
        <div>
          <label htmlFor="payload">Payload:</label>
          <div>
            <textarea name="payload" cols="30" rows="10" onChange={this.handlePayloadChange} />
          </div>
        </div>
        <div>
          <button type="submit">
            Submit
          </button>
        </div>
      </form>
    );
  }
}

WsForm.propTypes = {
  initAddScene: React.PropTypes.func.isRequired,
  queueId: React.PropTypes.string.isRequired
};

const mapStateToProps = state => ({
  queueId: getSelectedServerDetails(state).queue
});

const mapDispatchToProps = {
  initAddScene
};

export default connect(mapStateToProps, mapDispatchToProps)(WsForm);
