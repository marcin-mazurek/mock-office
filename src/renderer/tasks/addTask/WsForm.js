import React from 'react';
import { connect } from 'react-redux';
import { init as initAddTask } from './actions';
import { getSelected } from '../../servers/selectors';

class WsForm extends React.Component {
  constructor(props) {
    super(props);

    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleIntervalChange = this.handleIntervalChange.bind(this);
    this.handleBlockingChange = this.handleBlockingChange.bind(this);
    this.handleRequirementsChange = this.handleRequirementsChange.bind(this);
    this.handlePayloadChange = this.handlePayloadChange.bind(this);
    this.handleDelayChange = this.handleDelayChange.bind(this);
    this.state = {};
  }

  handleIntervalChange(event) {
    this.setState({
      interval: parseInt(event.target.value, 10)
    });
  }

  handleBlockingChange(event) {
    this.setState({
      blocking: event.target.checked
    });
  }

  handleRequirementsChange(event) {
    let requirements;

    try {
      requirements = JSON.parse(event.target.value);
    } catch (error) {
      // eslint-disable-next-line no-console
      console.error(error.message);
    }

    this.setState({
      requirements
    });
  }

  handlePayloadChange(event) {
    this.setState({
      taskPayload: event.target.value
    });
  }

  handleDelayChange(event) {
    this.setState({
      delay: parseInt(event.target.value, 10)
    });
  }

  handleSubmit(event) {
    event.preventDefault();
    this.props.initAddTask(this.props.serverId, this.state);
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <fieldset>
          <div>
            <label htmlFor="interval">Interval:</label>
            <input type="number" name="interval" onChange={this.handleIntervalChange} />
          </div>
          <div>
            <label htmlFor="blocking">Blocking:</label>
            <input
              type="checkbox"
              name="blocking"
              onChange={this.handleBlockingChange}
              value={this.state.blocking}
            />
          </div>
          <div>
            <label htmlFor="delay">Delay:</label>
            <input type="number" name="delay" onChange={this.handleDelayChange} />
          </div>
          <div>
            <label htmlFor="requirements">Requirements:</label>
            <div>
              <textarea
                name="requirements"
                cols="30"
                rows="10"
                onChange={this.handleRequirementsChange}
              />
            </div>
          </div>
          <div>
            <label htmlFor="payload">Payload:</label>
            <div>
              <textarea name="payload" cols="30" rows="10" onChange={this.handlePayloadChange} />
            </div>
          </div>
          <div>
            <button type="submit">
              Submit
            </button>
          </div>
        </fieldset>
      </form>
    );
  }
}

WsForm.propTypes = {
  initAddTask: React.PropTypes.func.isRequired,
  serverId: React.PropTypes.string.isRequired
};

const mapStateToProps = state => ({
  serverId: getSelected(state)
});

const mapDispatchToProps = {
  initAddTask
};

export default connect(mapStateToProps, mapDispatchToProps)(WsForm);
