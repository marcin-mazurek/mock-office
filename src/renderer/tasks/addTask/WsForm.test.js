import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { WsForm } from './WsForm';

describe('WsForm', () => {
  test('default snapshot', () => {
    const props = {
      initAddTask: () => {},
      queueId: 'some id'
    };
    const wrapper = shallow(<WsForm {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
