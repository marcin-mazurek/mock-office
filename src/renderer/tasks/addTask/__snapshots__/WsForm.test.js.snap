exports[`WsForm  should render 1`] = `
<form
  onSubmit={[Function]}>
  <div>
    <label
      htmlFor="interval">
      Interval:
    </label>
    <input
      name="interval"
      onChange={[Function]}
      type="number" />
  </div>
  <div>
    <label
      htmlFor="blocking">
      Blocking:
    </label>
    <input
      name="blocking"
      onChange={[Function]}
      type="checkbox" />
  </div>
  <div>
    <label
      htmlFor="delay">
      Delay:
    </label>
    <input
      name="delay"
      onChange={[Function]}
      type="number" />
  </div>
  <div>
    <label
      htmlFor="requirements">
      Requirements:
    </label>
    <div>
      <textarea
        cols="30"
        name="requirements"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <label
      htmlFor="payload">
      Payload:
    </label>
    <div>
      <textarea
        cols="30"
        name="payload"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <button
      type="submit">
      Submit
    </button>
  </div>
</form>
`;

exports[`WsForm de 1`] = `
<form
  onSubmit={[Function]}>
  <div>
    <label
      htmlFor="interval">
      Interval:
    </label>
    <input
      name="interval"
      onChange={[Function]}
      type="number" />
  </div>
  <div>
    <label
      htmlFor="blocking">
      Blocking:
    </label>
    <input
      name="blocking"
      onChange={[Function]}
      type="checkbox" />
  </div>
  <div>
    <label
      htmlFor="delay">
      Delay:
    </label>
    <input
      name="delay"
      onChange={[Function]}
      type="number" />
  </div>
  <div>
    <label
      htmlFor="requirements">
      Requirements:
    </label>
    <div>
      <textarea
        cols="30"
        name="requirements"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <label
      htmlFor="payload">
      Payload:
    </label>
    <div>
      <textarea
        cols="30"
        name="payload"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <button
      type="submit">
      Submit
    </button>
  </div>
</form>
`;

exports[`WsForm defau 1`] = `
<form
  onSubmit={[Function]}>
  <div>
    <label
      htmlFor="interval">
      Interval:
    </label>
    <input
      name="interval"
      onChange={[Function]}
      type="number" />
  </div>
  <div>
    <label
      htmlFor="blocking">
      Blocking:
    </label>
    <input
      name="blocking"
      onChange={[Function]}
      type="checkbox" />
  </div>
  <div>
    <label
      htmlFor="delay">
      Delay:
    </label>
    <input
      name="delay"
      onChange={[Function]}
      type="number" />
  </div>
  <div>
    <label
      htmlFor="requirements">
      Requirements:
    </label>
    <div>
      <textarea
        cols="30"
        name="requirements"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <label
      htmlFor="payload">
      Payload:
    </label>
    <div>
      <textarea
        cols="30"
        name="payload"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <button
      type="submit">
      Submit
    </button>
  </div>
</form>
`;

exports[`WsForm default  1`] = `
<form
  onSubmit={[Function]}>
  <div>
    <label
      htmlFor="interval">
      Interval:
    </label>
    <input
      name="interval"
      onChange={[Function]}
      type="number" />
  </div>
  <div>
    <label
      htmlFor="blocking">
      Blocking:
    </label>
    <input
      name="blocking"
      onChange={[Function]}
      type="checkbox" />
  </div>
  <div>
    <label
      htmlFor="delay">
      Delay:
    </label>
    <input
      name="delay"
      onChange={[Function]}
      type="number" />
  </div>
  <div>
    <label
      htmlFor="requirements">
      Requirements:
    </label>
    <div>
      <textarea
        cols="30"
        name="requirements"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <label
      htmlFor="payload">
      Payload:
    </label>
    <div>
      <textarea
        cols="30"
        name="payload"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <button
      type="submit">
      Submit
    </button>
  </div>
</form>
`;

exports[`WsForm default s 1`] = `
<form
  onSubmit={[Function]}>
  <div>
    <label
      htmlFor="interval">
      Interval:
    </label>
    <input
      name="interval"
      onChange={[Function]}
      type="number" />
  </div>
  <div>
    <label
      htmlFor="blocking">
      Blocking:
    </label>
    <input
      name="blocking"
      onChange={[Function]}
      type="checkbox" />
  </div>
  <div>
    <label
      htmlFor="delay">
      Delay:
    </label>
    <input
      name="delay"
      onChange={[Function]}
      type="number" />
  </div>
  <div>
    <label
      htmlFor="requirements">
      Requirements:
    </label>
    <div>
      <textarea
        cols="30"
        name="requirements"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <label
      htmlFor="payload">
      Payload:
    </label>
    <div>
      <textarea
        cols="30"
        name="payload"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <button
      type="submit">
      Submit
    </button>
  </div>
</form>
`;

exports[`WsForm default sna 1`] = `
<form
  onSubmit={[Function]}>
  <div>
    <label
      htmlFor="interval">
      Interval:
    </label>
    <input
      name="interval"
      onChange={[Function]}
      type="number" />
  </div>
  <div>
    <label
      htmlFor="blocking">
      Blocking:
    </label>
    <input
      name="blocking"
      onChange={[Function]}
      type="checkbox" />
  </div>
  <div>
    <label
      htmlFor="delay">
      Delay:
    </label>
    <input
      name="delay"
      onChange={[Function]}
      type="number" />
  </div>
  <div>
    <label
      htmlFor="requirements">
      Requirements:
    </label>
    <div>
      <textarea
        cols="30"
        name="requirements"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <label
      htmlFor="payload">
      Payload:
    </label>
    <div>
      <textarea
        cols="30"
        name="payload"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <button
      type="submit">
      Submit
    </button>
  </div>
</form>
`;

exports[`WsForm default snap 1`] = `
<form
  onSubmit={[Function]}>
  <div>
    <label
      htmlFor="interval">
      Interval:
    </label>
    <input
      name="interval"
      onChange={[Function]}
      type="number" />
  </div>
  <div>
    <label
      htmlFor="blocking">
      Blocking:
    </label>
    <input
      name="blocking"
      onChange={[Function]}
      type="checkbox" />
  </div>
  <div>
    <label
      htmlFor="delay">
      Delay:
    </label>
    <input
      name="delay"
      onChange={[Function]}
      type="number" />
  </div>
  <div>
    <label
      htmlFor="requirements">
      Requirements:
    </label>
    <div>
      <textarea
        cols="30"
        name="requirements"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <label
      htmlFor="payload">
      Payload:
    </label>
    <div>
      <textarea
        cols="30"
        name="payload"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <button
      type="submit">
      Submit
    </button>
  </div>
</form>
`;

exports[`WsForm default snapshot 1`] = `
<form
  onSubmit={[Function]}>
  <div>
    <label
      htmlFor="interval">
      Interval:
    </label>
    <input
      name="interval"
      onChange={[Function]}
      type="number" />
  </div>
  <div>
    <label
      htmlFor="blocking">
      Blocking:
    </label>
    <input
      name="blocking"
      onChange={[Function]}
      type="checkbox" />
  </div>
  <div>
    <label
      htmlFor="delay">
      Delay:
    </label>
    <input
      name="delay"
      onChange={[Function]}
      type="number" />
  </div>
  <div>
    <label
      htmlFor="requirements">
      Requirements:
    </label>
    <div>
      <textarea
        cols="30"
        name="requirements"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <label
      htmlFor="payload">
      Payload:
    </label>
    <div>
      <textarea
        cols="30"
        name="payload"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <button
      type="submit">
      Submit
    </button>
  </div>
</form>
`;

exports[`test  should render 1`] = `
<form
  onSubmit={[Function]}>
  <div>
    <label
      htmlFor="interval">
      Interval:
    </label>
    <input
      name="interval"
      onChange={[Function]}
      type="number" />
  </div>
  <div>
    <label
      htmlFor="blocking">
      Blocking:
    </label>
    <input
      name="blocking"
      onChange={[Function]}
      type="checkbox" />
  </div>
  <div>
    <label
      htmlFor="delay">
      Delay:
    </label>
    <input
      name="delay"
      onChange={[Function]}
      type="number" />
  </div>
  <div>
    <label
      htmlFor="requirements">
      Requirements:
    </label>
    <div>
      <textarea
        cols="30"
        name="requirements"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <label
      htmlFor="payload">
      Payload:
    </label>
    <div>
      <textarea
        cols="30"
        name="payload"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <button
      type="submit">
      Submit
    </button>
  </div>
</form>
`;

exports[`test WsForm should render 1`] = `
<form
  onSubmit={[Function]}>
  <div>
    <label
      htmlFor="interval">
      Interval:
    </label>
    <input
      name="interval"
      onChange={[Function]}
      type="number" />
  </div>
  <div>
    <label
      htmlFor="blocking">
      Blocking:
    </label>
    <input
      name="blocking"
      onChange={[Function]}
      type="checkbox" />
  </div>
  <div>
    <label
      htmlFor="delay">
      Delay:
    </label>
    <input
      name="delay"
      onChange={[Function]}
      type="number" />
  </div>
  <div>
    <label
      htmlFor="requirements">
      Requirements:
    </label>
    <div>
      <textarea
        cols="30"
        name="requirements"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <label
      htmlFor="payload">
      Payload:
    </label>
    <div>
      <textarea
        cols="30"
        name="payload"
        onChange={[Function]}
        rows="10" />
    </div>
  </div>
  <div>
    <button
      type="submit">
      Submit
    </button>
  </div>
</form>
`;
